identity/identity.py

```diff
@@ -0,0 +1,161 @@
+    """
+    Perform the initial authentication (attest) with Platform using the Client ID
+    and Secret from DOT
+    """
+
+    response = requests.post(
+        ATTEST_URL, auth=(accounts_settings.CLIENT_ID, accounts_settings.CLIENT_SECRET)
+    )
+    if response.status_code == 200:
+        content = response.json()
+        container.access_jwt = jwt.JWT(
+            key=container.platform_jwks, jwt=content["access"]
+        )
+        container.refresh_jwt = jwt.JWT(
+            key=container.platform_jwks, jwt=content["refresh"]
+        )
+        return True
+    return False
+
+
+def validate_urn(urn):
+    """
+    Validate an urn to ensure it follows the specification we use in Penn Labs.
+    Use the format `urn:<organization>:<product slug or wildcard>`
+    Ex. `urn:pennlabs:platform` or `urn:pennlabs:*`.
+    """
+    # Matches urn:<organization>:<product or wildcard>
+    pattern = re.compile(r"^urn:[a-z-]+:(?:[a-z-]+|\*)$")
+    if not pattern.match(urn):
+        raise ImproperlyConfigured(f"Invalid urn: '{urn}'")
+
+
+def get_validated_claims(token):
+    """
+    Validates JWT and returns the claims if validated, None otherwise.
+    """
+    try:
+        validated_jwt = jwt.JWT(key=container.platform_jwks, jwt=token)
+        claims = json.loads(validated_jwt.claims)
+        return (
+            claims
+            if "use" in claims and claims["use"] == "access" and "sub" in claims
+            else None
+        )
+    except ValueError:
+        # Catches error if token has unrecognizable format
+        return None
+
+
+def _refresh_if_outdated():
+    """
+    Refresh the access jwt if it is expired.
+    """
+    access_claims = json.loads(container.access_jwt.claims)
+    # only continue if our access jwt is expired (with a 30 second buffer)
+    if time.time() < access_claims["exp"] - 30:
+        return
+
+    auth_headers = {"Authorization": f"Bearer {container.refresh_jwt.serialize()}"}
+    response = requests.post(REFRESH_URL, headers=auth_headers)
+    if response.status_code == 200:
+        content = response.json()
+        container.access_jwt = jwt.JWT(
+            key=container.platform_jwks, jwt=content["access"]
+        )
+    else:
+        if not attest():  # If attest fails
+            raise Exception("Cannot authenticate with platform")
+
+
+def authenticated_b2b_request(
+    method,
+    url,
+    params=None,
+    data=None,
+    headers=None,
+    cookies=None,
+    files=None,
+    timeout=None,
+    allow_redirects=True,
+    proxies=None,
+    hooks=None,
+    stream=None,
+    verify=None,
+    cert=None,
+    json=None,
+):
+    """
+    Helper method to make an authenticated b2b request NOTE be ABSOLUTELY sure you
+    only make a request to Penn Labs products, otherwise you will expose credentials
+    and bad things will happen
+    """
+
+    # Attempt refresh
+    _refresh_if_outdated()
+
+    # Update Headers
+    headers = {} if headers is None else headers
```

Comment:
headers = headers or {}

Reply:
Good catch! Just updated the default value of headers